suite-native/notifications/src/components/TransactionNotification.tsx

```diff
@@ -1,92 +1,136 @@
+    'tx-received': {
+        title: 'Incoming transaction',
+        prefix: 'from',
+        transactionType: 'recv',
+        isIconAnimated: true,
+    },
+    'tx-confirmed': {
+        title: 'Received transaction',
+        prefix: 'from',
+        transactionType: 'recv',
+        isIconAnimated: false,
+    },
+    'tx-sent': {
+        title: 'Sending transaction',
+        prefix: 'to',
+        transactionType: 'sent',
+        isIconAnimated: true,
+    },
+} as const satisfies Record<TransactionNotificationType, TransactionTypeProperties>;
 
 export const TransactionNotification = ({
-    txid,
-    accountKey,
+    notificationId,
     isHiddenAutomatically = true,
 }: TransactionNotificationProps) => {
     const navigation =
         useNavigation<StackNavigationProps<RootStackParamList, RootStackRoutes.AppTabs>>();
+    const dispatch = useDispatch();
+
+    const notification = useSelector((state: NotificationsRootState) =>
+        selectTransactionNotificationById(state, notificationId),
+    );
+
+    const accountKey = useSelector((state: AccountsRootState) =>
+        selectAccountKeyByDescriptorAndNetworkSymbol(
+            state,
+            notification?.descriptor,
+            notification?.symbol,
+        ),
+    );
+
+    const txid = notification?.txid ?? '';
 
     const transaction = useSelector((state: TransactionsRootState) =>
-        selectTransactionByTxidAndAccountKey(state, txid, accountKey),
+        selectTransactionByTxidAndAccountKey(state, txid, accountKey ?? ''),
     );
 
     const transactionTargetAddress = useSelector((state: TransactionsRootState) =>
-        selectTransactionFirstTargetAddress(state, txid, accountKey),
+        selectTransactionFirstTargetAddress(state, txid, accountKey ?? ''),
+    );
+
+    const handleRemoveNotification = useCallback(
+        () => dispatch(notificationsActions.close(notificationId)),
+        [dispatch, notificationId],
     );
 
-    if (!transaction) return null;
+    if (!accountKey || !notification || !transactionTargetAddress) return null;
 
-    const { title, prefix } = transactionTypeToContentMap[transaction.type];
+    const { title, prefix, transactionType, isIconAnimated } =
+        transactionTypeToContentMap[notification.type];
 
     const navigateToTransactionDetail = () => {
         navigation.navigate(RootStackRoutes.TransactionDetail, {
-            txid: transaction.txid,
+            txid,
             accountKey,
         });
+        handleRemoveNotification();
     };
 
     return (
         <Notification
             isHiddenAutomatically={isHiddenAutomatically}
+            onHide={handleRemoveNotification}
             onPress={navigateToTransactionDetail}
             title={title}
             description={
                 <TransactionNotificationDescription
-                    networkSymbol={transaction.symbol}
-                    amount={transaction.amount}
+                    amount={transaction?.amount}
```

Comment:
Then you don't have to worry about fallback in description component. Undefined shouldn't be acceptable as prop 

```suggestion
                    amount={transaction?.amount ?? null}
```

Reply:
114b78127